### PR TITLE
chore: Create main.yml

### PR DESCRIPTION
**Pull request by software-dov**

### CODE REVIEW

**Comment by busunkim96** [comment]
Whenever the dust has cleared, please remember to edit https://github.com/googleapis/gapic-generator-python/blob/master/.github/sync-repo-settings.yaml so repo settings bot doesn't trample the required checks. :)

File: .github/sync-repo-settings.yaml [hunk 1]
```diff
@@ -5,24 +5,6 @@ branchProtectionRules:
 - pattern: master
   isAdminEnforced: true
   requiredStatusCheckContexts:
-    - 'ci/circleci: docs'
```
Comment (alexander-fenster) [inline .github/sync-repo-settings.yaml:8]: should we add the actions CI here?

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,331 @@
+# WIP github actions
```
Comment (alexander-fenster) [inline .github/workflows/tests.yaml:1]: still WIP? :)

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+name: Tests
+
+# Controls when the action will run.
+on:
+  pull_request:
+    branches: [ master ]
```
Comment (alexander-fenster) [inline .github/workflows/tests.yaml:7]: If you use `$default-branch` here it won't break when we eventually switch to `main`

File: .github/sync-repo-settings.yaml (hunk 1 above)
Comment (software-dov) [reply .github/sync-repo-settings.yaml:8]: I was under the impression that sync-repo-settings.yaml was generated based on settings saved in the UI and was not intended to be hand edited. @busunkim96 can you confirm or correct?

**Review by alexander-fenster — APPROVED** [review]
Wow, this is one of the most complicated CI setups I've seen in our org.
Thanks for doing it @software-dov!

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (software-dov) [reply .github/workflows/tests.yaml:1]: I mean...kind of? Publish package I've deferred deliberately, and publish_image is commented out as well. I'll remove the comment, though.

File: .github/workflows/tests.yaml (hunk 3 above)
Comment (software-dov) [reply .github/workflows/tests.yaml:7]: Good catch.

File: .github/sync-repo-settings.yaml (hunk 1 above)
Comment (busunkim96) [reply .github/sync-repo-settings.yaml:8]: @alexander-fenster is correct, the names of the new actions CI should be added here so sync-repo-settings bot preserves it as a required checks. 

The file is added to repos that have non-default CI configurations and overrides the default settings.

File: .github/workflows/tests.yaml [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+            unzip protoc-3.12.1.zip
+            sudo ln -s /usr/src/protoc/bin/protoc /usr/local/bin/protoc
+      - name: Run showcase tests.
+        run: nox -s ${{ matrix.target }}
+  showcase-mtls:
+    strategy:
+      matrix:
+        target: [showcase_mtls, showcase_mtls_alternative_templates]
+    # needs:
+    #   - [docs, mypy, unit, showcase-unit, showcase-mypy]
```
Comment (busunkim96) [inline .github/workflows/tests.yaml:112]: Should this be deleted or is it a palceholder?

**Review by busunkim96 — APPROVED** [review]

File: .github/workflows/tests.yaml (hunk 4 above)
Comment (software-dov) [reply .github/workflows/tests.yaml:112]: It's just cruft. Removing.

File: .github/sync-repo-settings.yaml (hunk 1 above)
Comment (software-dov) [reply .github/sync-repo-settings.yaml:8]: Done

**Pull request merged**